src/Orleans.Runtime.Abstractions/Hosting/ISiloBuilder.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Hosting
+{
+    /// <summary>
+    /// Functionality for building <see cref="ISiloHost"/> instances.
+    /// </summary>
+    public interface ISiloBuilder
```

Comment:
>ISiloBuilder [](start = 21, length = 12)

What if we just made the SiloHostBuilder implement this? I suppose it's done this way to avoid a breaking change and allow chaining calls that return ISiloHostBuilder, but is it worth it? or perhaps mark ISiloHostBuilder as obsolete and soon get rid of it. Adding duplicate extension methods for both ISiloBuilder and iSiloHostBuilder (triplicate if you count IServiceCollection, and then all the combinations that pass the context too) is a tax not just for the core team, but for every extension developer out there, and easy to get wrong.

Reply:
Avoiding breakages is precisely the reason for doing it this way. We could also have implemented the extensions using generic parameters to return the same type as passed in and avoided duplication but that is more complicated and degrades the experience for methods which have non-inferred generic parameters (since the usr would also have to specify the silo builder type parameter).

We should remove the IServiceCollection parameters and consider obsoleting the ISHB bits soon